src/main/java/de/tum/in/www1/artemis/web/rest/AchievementResource.java

```diff
@@ -0,0 +1,130 @@
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.*;
+
+import de.tum.in.www1.artemis.domain.Achievement;
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.repository.AchievementRepository;
+import de.tum.in.www1.artemis.service.AchievementService;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.service.UserService;
+import de.tum.in.www1.artemis.web.rest.errors.AccessForbiddenException;
+import de.tum.in.www1.artemis.web.rest.util.HeaderUtil;
+
+/**
+ * REST controller for managing Achievements.
+ */
+@RestController
+@RequestMapping("/api")
+public class AchievementResource {
+
+    private static final String ENTITY_NAME = "achievement";
+
+    @Value("${jhipster.clientApp.name}")
+    private String applicationName;
+
+    private final Logger log = LoggerFactory.getLogger(AchievementResource.class);
+
+    private final AuthorizationCheckService authCheckService;
+
+    private final AchievementService achievementService;
+
+    private final UserService userService;
+
+    private final AchievementRepository achievementRepository;
+
+    public AchievementResource(AuthorizationCheckService authCheckService, AchievementService achievementService, UserService userService,
+            AchievementRepository achievementRepository) {
+        this.authCheckService = authCheckService;
+        this.achievementService = achievementService;
+        this.userService = userService;
+        this.achievementRepository = achievementRepository;
+    }
+
+    /**
+     * GET /achievements : Get all achievements for user.
+     *
+     * @return the ResponseEntity with status 200 (OK) and with the found achievements as body
+     */
+    @GetMapping("/achievements")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Set<Achievement>> getAchievementsForUser() {
+        User user = userService.getUser();
+        log.debug("REST request to get achievements for user : {}", user.getLogin());
+        Set<Achievement> achievements = achievementService.findAllForUser(user.getId());
+        return ResponseEntity.ok(achievements);
+    }
+
+    /**
+     * GET /courses/{courseId}/achievements : Get all achievements for course.
+     *
+     * @param courseId the course to which the achievements belong to
+     * @return the ResponseEntity with status 200 (OK) and with the found achievements as body
+     */
+    @GetMapping("/courses/{courseId}/achievements")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Set<Achievement>> getAchievementsForCourse(@PathVariable Long courseId) {
+        log.debug("REST request to get achievements for course : {}", courseId);
+        Set<Achievement> achievements = achievementService.findAllForCourse(courseId);
```

Comment:
Not really. Well achievements do have a relation to users but users are loaded lazily within achievements by default and since not requested explicitly they remain null here.